### PR TITLE
RocketChat: Initial Role Addition

### PR DESCRIPTION
**Pull request by Minds3t**
Basically translated the official docker-compose.yml into ansible and to work with cloudbox.

Added role dir and main.yml
Added role to community.yml
Added role to appveyor.yml

### CODE REVIEW

File: roles/rocketchat/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    - "/opt/rocketchat"
+    
+- name: Create and start RocketChat container
+  docker_container:
+    name: "rocketchat"
+    image: "rocketchat/rocket.chat:latest"
+    command: bash -c 'for i in `seq 1 30`; do node main.js && s=$$? && break || s=$$?; echo "Tried $$i times. Waiting 5 secs..."; sleep 5; done; (exit $$s)'
+    pull: yes
+    published_ports:
+      - "3000:3000"
```
Comment (desimaniac) [inline roles/rocketchat/tasks/main.yml]: Does this need the port to be exposed? Does it not work behind reverse proxy?

File: roles/rocketchat/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+- name: Stop and remove any existing container
+  docker_container: "name={{ item }} state=absent"
+  with_items:
+    - rocketchat
+    
+- name: Create required directories
+  file: "path={{ item }} state=directory mode=0775 owner={{ user.name }} group={{ user.name }}"
+  with_items:
+    - "/opt/rocketchat"
```
Comment (desimaniac) [inline roles/rocketchat/tasks/main.yml]: Not needed.

File: roles/rocketchat/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+- name: Stop and remove any existing container
+  docker_container: "name={{ item }} state=absent"
+  with_items:
+    - rocketchat-mongo
+
+- name: Create required directories
+  file: "path={{ item }} state=directory mode=0775 owner={{ user.name }} group={{ user.name }}"
+  with_items:
+    - "/opt/rocketchat/data/db"
+    - "/opt/rocketchat/data/dump"
```
Comment (desimaniac) [inline roles/rocketchat/tasks/main.yml]: Would prob add /opt/rocketchat/uploads here.

**Comment by Minds3t** [comment]
I have applied your recommendations

File: roles/rocketchat/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  include_role:
+    name: cloudflare-dns
+  vars:
+    subdomain: rocketchat
+  when: cloudflare_enabled
+
+- name: Stop and remove any existing container
+  docker_container: "name={{ item }} state=absent"
+  with_items:
+    - rocketchat
```
Comment (desimaniac) [inline roles/rocketchat/tasks/main.yml]: We can move this right under line 16. 

Could prob avoid any issues with the two previous container tasks by shutting down and removing an existing rocketchat container.

**Comment by desimaniac** [comment]
Looks like AppVeyor is having issues, so we can just merge after this one change.

**Comment by Minds3t** [comment]
Now it should be fine

**Pull request merged**